src/linux-memory.c

```diff
@@ -1253,13 +1253,13 @@ generic_get_memory_info( PAPI_hw_info_t *hw_info )
             MEMDBG("Could not read cache type\n");
             goto unrecoverable_error;
         }
-        if (!strcmp(type_string,"Data")) {
+        if (!strcmp(type_string,"Data") || !strcmp(type_string,"Data\n")) {
```

Comment:
Instead of using multiple strcmp() calls for each IF, I would suggest using one strncmp() instead. You know how long "data" and "Instruction", etc. are.